### ISSUE TITLE
Hibernate Reactive Panache MySQL 

### ISSUE THREAD

**Issue by douglasss** [quarkusio/quarkus#35568]
### Describe the bug

After Calling find() and count() on a table few times with ~10rows I am getting timeout:
I believe the connections are not being closed after the calls.

2023-08-25 11:28:34,464 ERROR [com.tha.ota.exc.ThrowableMapper] (vert.x-eventloop-thread-3) Timeout: io.vertx.core.impl.NoStackTraceThrowable: Timeout

```java
public static Uni<PageResult<Table>> findPage(Set<String> param1,
                                                       Param2 param2,
                                                       ...,
                                                       Page page,
                                                       Sort sort) {

        Set<String> whereClauses = new HashSet<>();
        Map<String, Object> params = new HashMap<>();
        
        ... I Build the where clause here

        Uni<Long> count = countBy(whereClause, params);

        Uni<List<Table>> content = findAllBy(whereClause, params, page, sort);

        return Uni.combine().all().unis(count, content)
                .asTuple()
                .map(tuple -> new PageResult<>(page.index, page.size, tuple.getItem1(), tuple.getItem2()));
    }

    private static Uni<Long> countBy(String whereClause, Map<String, Object> params) {
        return Table.count(
                "from Table t join t.table2 t2 join t2.table3 t3" + whereClause,
                params);
    }

    public static Uni<List<Table>> findAllBy(String whereClause, Map<String, Object> params, Page page, Sort sort) {
        String query = "from Table t join fetch t.table2 t2 join fetch t2.table3 t3" + whereClause;
        return Table.<Table>find(
                        query,
                        sort,
                        params
                )
                .page(page)
                .list();
    }
```

### Expected behavior

Not getting I timeout exception

### Output of `java -version`

OpenJDK Runtime Environment Corretto-17.0.7.7.1 (build 17.0.7+7-LTS)

### Quarkus version or git rev

3.3.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.2

**Comment by DavideD** [comment]
>  return Uni.combine().all().unis(count, content)

Hibernate Reactive doesn't support `combine().all()`. That's probably what's causing the TimeOut

**Comment by douglasss** [comment]
@DavideD Do you know another away to do that? I've tried something with `flatMap` but I got the same timout.

```java
count.flatMap(countResult -> content.map(contentResult -> new PageResult<>(page.index, page.size, countResult, contentResult)));
```

**Comment by douglasss** [comment]
@DavideD I create a small test project here (https://github.com/douglasss/panache-reactive-timeout)

If you run the tests with `quarkus test`, you can see that it repeats itself for 21 times.
At the 21th attempt we have the `io.vertx.core.impl.NoStackTraceThrowable: Timeout`

<img width="1792" alt="Captura de Tela 2023-08-25 às 15 24 27" src="https://github.com/quarkusio/quarkus/assets/5478159/d6ed965a-e319-4a36-b338-73a7274103d1">

**Comment by DavideD** [comment]
Thanks @douglasss, that's great!
I will try to have a look at it today

**Comment by DavideD** [comment]
I tried the project and adding `@WithSession` seems to solve the issue:

```java
        @Get
	@WithSession
	public Uni<PageResult<Product>> getAll() {
		return find();
	}
```

I would have expected some error when the annotation is missing though.
@mkouba any insight?

**Comment by mkouba** [comment]
TLDR> There is a bug in the implementation of `@WithSessionOnDemand`. `@WithSession` and `@WithTransaction` indeed work correctly.

Every stage that calls a Panache entity method (e.g. `PanacheEntityBase.count()`) opens a new reactive session, but only the last one is closed. I'll try to send a PR with a fix shortly.

**Comment by douglasss** [comment]
@DavideD and @mkouba, Thanks for getting back so fast!
Both suggestions (WithSession and nested calls) worked perfectly on our project!
